skada/metrics.py

```diff
@@ -631,7 +631,13 @@ def _score(self, estimator, X, y, sample_domain=None, **params):
 
             if y_type == Y_Type.DISCRETE:
                 # We go back to the original labels
-                y_pred_source = le.inverse_transform(y_pred_source)
+                try:
+                    y_pred_source = le.inverse_transform(y_pred_source)
+                except ValueError as e:
+                    raise ValueError(
+                        "CircularValidation: predicted source labels contain "
+                        "unknown labels not seen during label encoding."
+                    ) from e
```

Comment:
I prefer that you directly check the issue https://github.com/scikit-adaptation/skada/issues/245 with a `if ...`. The code right now might print
```
"CircularValidation: predicted source labels contain unknown labels not seen during label encoding."
```
for a totally unrelated problem.